File: R/box.R
## TODO: Maybe use 'cssText' instead of list
#' Add a (text) box to the map
#'
#' @inheritParams add_layer
#' @param content The text which will be displayed on the map.
#' @param position The position on the map.
#' @param style CSS options in the form of \emph{cssText} applied to the element.
#' @export
add_box <- function(map, content, position = "top-right", style = "") {
  invoke_method(map, "addBox", content, position, style)
}

File: javascript/src/box.js
export default function(content, position, style) {
  const map = this;
  const widgetElement = map.getContainer();
  const boxElement = document.createElement("div");
  // boxElement.className = "map-overlay";
  boxElement.className = `box box-${position || "top-right"}`;
  // Object.assign(boxElement.style, style);
  boxElement.style.cssText = style || "";

  boxElement.innerHTML = content;
  widgetElement.appendChild(boxElement);
}
